source/data-api/generated-endpoints.txt

```diff
@@ -0,0 +1,1364 @@
+            :emphasize-lines: 5
+            
+            curl -X POST \
+              https://data.mongodb-api.com/app/myapp-abcde/endpoint/data/v1/action/findOne \
+              -H 'api-key: <API Key>' \
+              -H 'Content-Type: application/json' \
+              -H 'Accept: application/json' \
+              --data-raw '{
+                 "dataSource": "mongodb-atlas",
+                 "database": "learn-data-api",
+                 "collection": "hello",
+                 "projection": { "greeting": 1, "date": 1 }
+              }'
+         
+         .. output::
+            :language: json
+            
+            {
+              "_id": "629971c0d71aad65bd59c545",
+              "greeting": "Hello, Leafie!",
+              "date": "2022-06-07T08:10:30.998Z"
+            }
+
+.. _data-api-authenticate:
+
+Authenticate the Request
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+To keep your data secure and correct, generated Data API endpoints
+always run in the context of a specific, registered user and
+automatically enforce your app's :ref:`rules <rules>` and :ref:`schemas
+<schemas>`. Incoming requests must include one or more headers that
+contain valid user credentials for one of the following enabled
+authentication providers:
+
+- :ref:`Email/Password <email-password-authentication>`
+- :ref:`API Key <api-key-authentication>`
+- :ref:`Custom JWT <custom-jwt-authentication>`
+
+.. tabs-realm-auth-providers::
+   
+   .. tab::
+      :tabid: local-userpass
+
+      .. code-block:: bash
+         :emphasize-lines: 3-4
+         
+         curl -X POST \
+           https://data.mongodb-api.com/app/myapp-abcde/endpoint/data/v1/action/findOne \
+           -H 'email: <Email Address>' \
+           -H 'password: <Password>' \
+           -H 'Content-Type: application/json' \
+           --data-raw '{
+             "dataSource": "mongodb-atlas",
+             "database": "learn-data-api",
+             "collection": "hello",
+           }'
+   
+   .. tab::
+      :tabid: api-key
+
+      .. code-block:: bash
+         :emphasize-lines: 3
+         
+         curl -X POST \
+           https://data.mongodb-api.com/app/myapp-abcde/endpoint/data/v1/action/findOne \
+           -H 'api-key: <API Key>' \
+           -H 'Content-Type: application/json' \
+           --data-raw '{
+             "dataSource": "mongodb-atlas",
+             "database": "learn-data-api",
+             "collection": "hello",
+           }'
+   
+   .. tab::
+      :tabid: custom-token
+
+      .. code-block:: bash
+         :emphasize-lines: 3
+         
+         curl -X POST \
+           https://data.mongodb-api.com/app/myapp-abcde/endpoint/data/v1/action/findOne \
+           -H 'jwtTokenString: <Custom JSON Web Token>' \
+           -H 'Content-Type: application/json' \
+           --data-raw '{
+             "dataSource": "mongodb-atlas",
+             "database": "learn-data-api",
+             "collection": "hello",
+           }'
+
+Endpoints
+---------
```

Comment:
as i was reading through the endpoint reference made me think that it'd be nice to have an openAPI spec for the data API. maybe future project?

Reply:
could replace this section w a openapi spec-based page

Reply:
I already have one mostly done actually! For the sake of short term expediency I just copied these over from the Atlas docs.